### PR TITLE
fix: prevent data leakage in scaler/encoder fit_transform calls

### LINKED ISSUE #514: Bug: Data leakage — scaler/encoder fit_transform() called on full dataset before train/test split
## Bug Report

**Description**
In `Automated Financial Reporting with Deep Learning/app.py`, the `MinMaxScaler` is fitted on
the **entire dataset** (including what would be test data) before any train/test split. This
causes **data leakage** — the model indirectly sees test data statistics during training,
leading to artificially inflated accuracy metrics that won't hold in real-world use.

## Affected Files

| File | Line | Issue |
|------|------|-------|
| `Automated Financial Reporting with Deep Learning/app.py` | 16 | `scaler.fit_transform()` on entire df |
| `Engineering Career Role Prediction/app.py` | 32 | `le.fit_transform()` inside prediction function using training encoder on live input |

## Code (Buggy)
```python
scaler = MinMaxScaler()
data_scaled = scaler.fit_transform(df.drop('Date', axis=1))  # leaks test stats into training
```

## Fix
```python
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
scaler = MinMaxScaler()
X_train_scaled = scaler.fit_transform(X_train)   # fit ONLY on train
X_test_scaled = scaler.transform(X_test)          # transform test separately
```

## Severity
High — produces misleadingly optimistic model performance; this is a teaching repo so it
actively trains beginners with incorrect ML practices

### PR DESCRIPTION
**Pull request by Daksh7785**
- scaler.fit_transform() now called only on X_train, not full dataset
- X_test uses scaler.transform() only — no re-fitting on test data
- LabelEncoder refactored out of prediction function — fitted once at training
- Applies correct fit/transform split pattern across all affected files
- Updated CHANGELOG.md under [Unreleased]
- Closes #514

## Summary

Closes #

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Documentation update
- [ ] Refactor
- [ ] Tests
- [ ] Infra / CI

## Checklist

- [ ] I have read [CONTRIBUTING.md](../CONTRIBUTING.md)
- [ ] My code follows the project style (PEP 8 for Python, ESLint for TS)
- [ ] I have added/updated tests where relevant
- [ ] `pytest backend/tests/` passes locally
- [ ] I have not committed `.env` or any secrets
- [ ] I have updated documentation if needed

### CODE REVIEW

**Comment by SdSarthak** [comment]
This PR adds an 'Automated Financial Reporting with Deep Learning' Streamlit app to the AegisAI repository. This is a different project with no connection to the EU AI Act compliance platform. Closing as wrong repository.
